source/main.py

```diff
@@ -0,0 +1,253 @@
+        rendered += term.black_on_white(line) if index == i else line
+        rendered += '\n'
+
+    rendered += term.move_y(term.height - 3)
+    rendered += '    ' + (term.black_on_white if index == entries else str)('Add new')
+    return rendered
+
+
+def render_file_explorer(term: Terminal, current_dir: str, files: List[str], dirs: List[str], index: int = 0) -> str:
+    """Render file explorer
+
+    Args:
+        term (Terminal): blessed.Terminal object
+        current_dir (str): currently selected directory
+        files (List[str]): list of files
+        dirs (List[str]): list of directories
+        index (int, optional): index of selected file/dir in parent directory. Defaults to 0.
+
+    Returns:
+        str: [description]
+    """
+    rendered = ''
+    rendered += term.black_on_white(set_string_length('    ' + current_dir, term.width)) + '\n\n'
+
+    rendered += '\n'.join([
+        (term.black_on_blue if index == i else term.blue)
+        (set_string_length(dirs[i], term.width))
+        for i in range(len(dirs))
+    ]) + '\n'
+
+    rendered += '\n'.join([
+        (term.black_on_white if index - len(dirs) == i else str)
+        (set_string_length(files[i], term.width))
+        for i in range(len(files))
+    ]) + '\n'
+
+    return rendered
+
+
+def select_new_puzzle(term: Terminal, current_dir: str = os.getcwd()) -> str:
+    """Select image from local files
+
+    Args:
+        term (Terminal): blessed.Terminal objet
+        current_dir (str, optional): directory currently selected. Defaults to os.getcwd().
+
+    Returns:
+        str: image path
+    """
+    print(term.clear)
+    print(term.home)
+
+    dirs = ['    ../']
+    files = []
+
+    for file in os.listdir(current_dir):
+        if file.endswith(SUPPORTED_FILE_TYPES) and os.path.isfile(os.path.join(current_dir, file)):
+            files.append('    ' + file)
+        elif os.path.isdir(os.path.join(current_dir, file)):
+            dirs.append('    ' + file + '/')
+
+    index = 0
+    while 1:
+        print(term.clear)
+        print(term.home)
+        print(render_file_explorer(term, current_dir, files, dirs, index))
+        inp = term.inkey()
+        if inp.code == term.KEY_UP:
+            index -= 1
+        if inp.code == term.KEY_DOWN:
+            index += 1
+        if inp.code == term.KEY_ENTER:
+            break
+
+        if index < 0:
+            index = 0
+        if index > len(files) + len(dirs) - 1:
+            index = len(files) + len(dirs) - 1
+
+    if index < len(dirs):
+        return select_new_puzzle(term, os.path.abspath(os.path.join(current_dir, dirs[index][4:])))
+
+    path = os.path.abspath(os.path.join(current_dir, files[index - len(dirs) - 1][4:]))
+    return path
+
+
+def main(term: Terminal) -> None:
+    """Main function
+
+    Args:
+        term (Terminal): blessed.Terminal object
+    """
+    index = 0
+
+    while 1:
```

Comment:
Can we switch all of these to using `True`?
```suggestion
    while True:
```